### PR TITLE
Fixed passing null for trim is deprecated in Mage_Core_Model_Resource_Db_Abstract

### PR DESCRIPTION
**Pull request by kyrena**
### Description

This fix `passing null to parameter #1 ($string) of type string is deprecated` for `trim` with PHP 8.1 / 8.2 / 8.3.
**Probably** it's not the best way, but it works with PHP 7.2 / 7.3 / 7.4 / 8.0 / 8.1 / 8.2 / 8.3.

To reproduce, go to any products attributes set, and save.

See also this [discussion comment](https://github.com/OpenMage/magento-lts/discussions/3025#discussioncomment-5195497) and #3325.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Core/Model/Resource/Db/Abstract.php [hunk 1]
```diff
@@ -656,10 +657,10 @@ protected function _checkUnique(Mage_Core_Model_Abstract $object)
 
                 if (is_array($unique['field'])) {
                     foreach ($unique['field'] as $field) {
-                        $select->where($field . '=?', trim($data->getData($field)));
+                        $select->where($field . '=?', trim((string)$data->getData($field)));
```
Comment (kiatng) [inline app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:660]:
```suggestion
                        if ($data->getData($field) !== null) {
                            $select->where($field . '=?', trim($data->getData($field)));
                        }
```

File: app/code/core/Mage/Core/Model/Resource/Db/Abstract.php [hunk 2]
```diff
@@ -656,10 +657,10 @@ protected function _checkUnique(Mage_Core_Model_Abstract $object)
...
                     }
                 } else {
-                    $select->where($unique['field'] . '=?', trim($data->getData($unique['field'])));
+                    $select->where($unique['field'] . '=?', trim((string)$data->getData($unique['field'])));
```
Comment (kiatng) [inline app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:663]:
```suggestion
                } elseif ($data->getData($unique['field']) !== null) {
                    $select->where($unique['field'] . '=?', trim($data->getData($unique['field'])));
```

**Review by kiatng — COMMENTED** [review]
Am I right to think that if it's null, there is no need to check for uniqueness?

[edit]
I checked, unique column allows multiple nulls, but not multiple empty strings.

File: app/code/core/Mage/Core/Model/Resource/Db/Abstract.php (hunk 1 above)
Comment (luigifab) [reply app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:660]: It's probably better, I will try.

Comment (kyrena) [reply app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:660]: No this does not work.

File: app/code/core/Mage/Core/Model/Resource/Db/Abstract.php (hunk 2 above)
Comment (kyrena) [reply app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:663]: This is working when I save an attributes set, but are there side effects? I don't know.

File: app/code/core/Mage/Core/Model/Resource/Db/Abstract.php (hunk 1 above)
Comment (kiatng) [reply app/code/core/Mage/Core/Model/Resource/Db/Abstract.php:660]: I  tested this by saving an attribute set with an existing name, it works just fine.

> Attribute set with the "Default" name already exists.

What error do you get?

**Review by kiatng — APPROVED** [review]
Tested.

**Review by fballiano — APPROVED** [review]

**Pull request merged**